lib/persist.php

```diff
@@ -43,6 +43,34 @@ function inputIsValid($input) {
   return FALSE;
 }
 
+/**
+ * Recursively processes the given object and runs all its string-valued properties
+ * through `REDCap::escapeHtml`. The original object is mutated.
+ *
+ * @param object $obj - the object to sanitize
+ */
+function sanitizeStringProperties($obj) {
+  $props = get_object_vars($obj);
+  foreach ($props as $prop => $val) {
+    if (is_object($val)) {
+      sanitizeStringProperties($val);
+    } else if (is_string($val)) {
+      $obj->{$prop} = REDCap::filterHtml($val);
```

Comment:
Using `REDCap::filterHtml` rather than `REDCap::escapeHtml` because the latter breaks filter logic.